### PR TITLE
Fix attacker assumes bugged while npc has ish

### PR DESCRIPTION
**Pull request by ODAKAB**
Attacker thinks attack is bugged if npc hp doesn't drop but also we are in range(under 700)
Added npc ish check to improve hp check, tested and works.

### CODE REVIEW

File: src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ protected void tryAttackOrFix() {
             sendAttack(250, 5000, true);
         } else if (!hero.isAttacking(target) || !hero.isAiming(target)) {
             sendAttack(1500, 5000, false);
-        } else if (target.health.hpDecreasedIn(1500) || hero.locationInfo.distance(target) > 700) {
+        } else if ((target.health.hpDecreasedIn(1500) || target.hasEffect(16)) || hero.locationInfo.distance(target) > 700) {
```
Comment (Pablete1234) [inline src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java:113]: Avoid using magic numbers, use the effect constants when possible target.hasEffect(Effect.NPC_ISH)

**Review by Pablete1234 — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java (hunk 1 above)
Comment (ODAKAB) [reply src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java:113]: paranthesis was there to specify ish check is meant to work together with hp check which isn't related to distance, helps reader by being more understandable for future
if you agree i'll change to effect name only

**Comment by ODAKAB** [comment]
Logic exceeds one line after last change, can group it under a method isAttackValid or similar

File: src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java (hunk 1 above)
Comment (ODAKAB) [reply src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java]: https://github.com/darkbot-reloaded/DarkBot/pull/49/commits/22e09ec75c29ed4437f946bc19509667c9592bc1

**Review by Pablete1234 — APPROVED** [review]

**Pull request merged**